addons/hr_work_entry_holidays/models/hr_leave.py

```diff
@@ -19,32 +20,42 @@ class HrLeaveType(models.Model):
 class HrLeave(models.Model):
     _inherit = 'hr.leave'
 
+    def _compute_resource_calendar_id(self):
+        super()._compute_resource_calendar_id()
+        for leave in self.filtered(lambda l: l.employee_id):
+            # We use the request dates to find the contracts, because date_from
+            # and date_to are not set yet at this point. Since these dates are
+            # used to get the contracts for which these leaves apply and
+            # contract start- and end-dates are just dates (and not datetimes)
+            # these dates are comparable.
+            if leave.employee_id:
+                contracts = leave.employee_id.sudo()._get_contracts(leave.request_date_from,
+                                                                    leave.request_date_to or leave.request_date_from,
+                                                                    states=['open', 'close'])
+                if contracts:
+                    # If there are more than one contract they should all have the
+                    # same calendar, otherwise a constraint is violated.
+                    leave.resource_calendar_id = contracts[:1].resource_calendar_id or leave.resource_calendar_id
+
     def _create_resource_leave(self):
```

Comment:
This override seem to create additional resources if an employee takes some vacation on a period for which their contract that has a different calendar (one for the current contract and one for the other).
Prior to this commit, that would have meant having one resource for each calendar (which was already useless I guess), but with your commit it will create a second resource based on the same calendar. Overall, since the correct calendar will already be stored in the leave, that override method could be dropped.

Reply:
Ideally yes, but doing that seems to break some stuff. But I'm going to look into it.

Reply:
I fixed this (and some other stuff as well)